src/PKSim.Core/Mappers/ExpressionProfileToExpressionProfileBuildingBlockMapper.cs

```diff
@@ -0,0 +1,90 @@
+{
+   public interface IExpressionProfileToExpressionProfileBuildingBlockMapper : IMapper<ExpressionProfile, ExpressionProfileBuildingBlock>
+   {
+   }
+
+   public class ExpressionProfileToExpressionProfileBuildingBlockMapper : IExpressionProfileToExpressionProfileBuildingBlockMapper
+   {
+      private readonly IObjectBaseFactory _objectBaseFactory;
+      private readonly IEntityPathResolver _entityPathResolver;
+      private IFormulaFactory _formulaFactory;
+
+      public ExpressionProfileToExpressionProfileBuildingBlockMapper(IObjectBaseFactory objectBaseFactory, IEntityPathResolver entityPathResolver, IFormulaFactory formulaFactory)
+      {
+         _objectBaseFactory = objectBaseFactory;
+         _entityPathResolver = entityPathResolver;
+         _formulaFactory = formulaFactory;
+      }
+
+      public ExpressionProfileBuildingBlock MapFrom(ExpressionProfile expressionProfile)
+      {
+         var expressionProfileBuildingBlock = _objectBaseFactory.Create<ExpressionProfileBuildingBlock>();
+
+         expressionProfileBuildingBlock.Name = expressionProfile.Name;
+         expressionProfileBuildingBlock.PKSimVersion = ProjectVersions.Current.VersionDisplay;
+         expressionProfileBuildingBlock.Description = expressionProfile.Description;
+
+         setExpressionType(expressionProfile, expressionProfileBuildingBlock);
+
+         var allParameters = expressionProfile.GetAllChildren<IParameter>();
+
+         foreach (var parameter in allParameters)
+         {
+            var expressionParameter = mapExpressionParameterFromExpressionProfile(parameter, expressionProfileBuildingBlock);
+            expressionProfileBuildingBlock.Add(expressionParameter);
+         }
+
+         return expressionProfileBuildingBlock;
+      }
+
+      private ExpressionParameter mapExpressionParameterFromExpressionProfile(IParameter parameter,
+         ExpressionProfileBuildingBlock expressionProfileBuildingBlock)
+      {
+         var expressionParameter = _objectBaseFactory.Create<ExpressionParameter>();
+
+         if (parameter.Formula != null && parameter.Formula.IsCachable())
+         {
+            var formula = _formulaFactory.RateFor(CoreConstants.CalculationMethod.EXPRESSION_PARAMETERS, parameter.Formula.Name,
+               expressionProfileBuildingBlock.FormulaCache);
+            expressionParameter.Formula = formula;
+         }
+         else
+         {
+            (expressionParameter.StartValue, _) = parameter.TryGetValue();
+         }
+
+         expressionParameter.Name = parameter.Name;
+
+         expressionParameter.Path = _entityPathResolver.ObjectPathFor(parameter);
+         expressionParameter.Dimension = parameter.Dimension;
+         expressionParameter.DisplayUnit = parameter.DisplayUnit;
+         return expressionParameter;
+      }
+
+      private static void setExpressionType(ExpressionProfile expressionProfile, ExpressionProfileBuildingBlock expressionProfileBuildingBlock)
```

Comment:
no this is not what I meant sorry. While this, works, try to avoid side effects in methods

sthg like this

mapExpressionType(moleculeType){
   switch (moleculeType)
         {
            case QuantityType.Enzyme:
              return  ExpressionTypes.MetabolizingEnzyme;
            case QuantityType.Transporter:
               return ExpressionTypes.TransportProtein;
            default:
               expressionProfileBuildingBlock.Type = ExpressionTypes.ProteinBindingPartner;
               break;
         }
etc..
}

ALSO NO DEFAULT PLEASE. The default case should throw an exception because we are missing a type.